joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/CacheBuilder.scala

```diff
@@ -148,22 +153,28 @@ trait CacheBuilder(implicit withSchemaValidation: ValidationMode) { this: AstCre
   protected def processFuncDecl(
     funcDeclVal: Value
   ): (String, String, String, Value, Option[(String, String, String, ParserNodeInfo)], Map[String, List[String]]) = {
-    val name         = funcDeclVal(ParserKeys.Name).obj(ParserKeys.Name).str
-    val receiverInfo = getReceiverInfo(Try(funcDeclVal(ParserKeys.Recv)))
-    val methodFullname = receiverInfo match
-      case Some(_, typeFullName, _, _) =>
-        s"$typeFullName.$name"
-      case _ =>
-        s"$fullyQualifiedPackage.$name"
-    // TODO: handle multiple return type or tuple (int, int)
-    val genericTypeMethodMap = processTypeParams(funcDeclVal(ParserKeys.Type))
-    val (returnTypeStr, _) =
-      getReturnType(funcDeclVal(ParserKeys.Type), genericTypeMethodMap).headOption
-        .getOrElse(("", null))
-    val params = funcDeclVal(ParserKeys.Type)(ParserKeys.Params)(ParserKeys.List)
-    val signature =
-      s"$methodFullname(${parameterSignature(params, genericTypeMethodMap)})$returnTypeStr"
-    goGlobal.recordFullNameToReturnType(methodFullname, returnTypeStr, signature)
-    (name, methodFullname, signature, params, receiverInfo, genericTypeMethodMap)
+    val name = funcDeclVal(ParserKeys.Name).obj(ParserKeys.Name).str
+    if (!goGlobal.processingDependencies || goGlobal.processingDependencies && name.headOption.exists(_.isUpper)) {
+      // Ignoring recording the method details when we are processing dependencies code with functions name starting with lower case letter
+      // As the functions starting with lower case letters will only be accessible within that package. Which means
+      // these methods / functions are not going to get referred from main source code.
+      val receiverInfo = getReceiverInfo(Try(funcDeclVal(ParserKeys.Recv)))
+      val methodFullname = receiverInfo match
+        case Some(_, typeFullName, _, _) =>
+          s"$typeFullName.$name"
+        case _ =>
+          s"$fullyQualifiedPackage.$name"
+      // TODO: handle multiple return type or tuple (int, int)
+      val genericTypeMethodMap = processTypeParams(funcDeclVal(ParserKeys.Type))
+      val (returnTypeStr, _) =
+        getReturnType(funcDeclVal(ParserKeys.Type), genericTypeMethodMap).headOption
+          .getOrElse(("", null))
+      val params = funcDeclVal(ParserKeys.Type)(ParserKeys.Params)(ParserKeys.List)
+      val signature =
+        s"$methodFullname(${parameterSignature(params, genericTypeMethodMap)})$returnTypeStr"
+      goGlobal.recordFullNameToReturnType(methodFullname, returnTypeStr, signature)
+      (name, methodFullname, signature, params, receiverInfo, genericTypeMethodMap)
+    } else
+      ("", "", "", Value("{}"), None, Map())
```

Comment:
When tuples start to get this long, I wonder if this shouldn't rather be a case class

Reply:
done